From 615ef7c75f5d7af2e80d7da6df43ee6eeb73c699 Mon Sep 17 00:00:00 2001
Message-Id: <615ef7c75f5d7af2e80d7da6df43ee6eeb73c699.1421835210.git.chang-joon.lee@intel.com>
In-Reply-To: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
References: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Mon, 1 Dec 2014 14:54:49 +0000
Subject: [PATCH 04/63] Revert "FOR_UPSTREAM [VPG]: drm/i915: capture more
 detail in ringbuf & LRC request error dump"

This reverts commit b3af8a51120fbc4b15e02be318acbacb5028db78.

Change-Id: I6b95d2ece2914501c529b98960b2a8c0d4a95b56
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h       |    8 +-------
 drivers/gpu/drm/i915/i915_gpu_error.c |   19 +++----------------
 2 files changed, 4 insertions(+), 23 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 2d90264..18e8b34 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -367,10 +367,6 @@ struct drm_i915_error_state {
 		int num_requests;
 
 		/* our own tracking of ring head and tail */
-		struct intel_context *ring_context;
-		int size;
-		int space;
-		u32 last_retired_head;
 		u32 cpu_ring_head;
 		u32 cpu_ring_tail;
 
@@ -402,11 +398,9 @@ struct drm_i915_error_state {
 		} *ringbuffer, *batchbuffer, *wa_batchbuffer, *ctx, *hws_page;
 
 		struct drm_i915_error_request {
-			struct intel_context *ctx;
 			long jiffies;
-			u32 head;
-			u32 tail;
 			u32 seqno;
+			u32 tail;
 		} *requests;
 
 		struct {
diff --git a/drivers/gpu/drm/i915/i915_gpu_error.c b/drivers/gpu/drm/i915/i915_gpu_error.c
index 2b282c7..20ea5e4 100644
--- a/drivers/gpu/drm/i915/i915_gpu_error.c
+++ b/drivers/gpu/drm/i915/i915_gpu_error.c
@@ -300,11 +300,6 @@ static void i915_ring_error_state(struct drm_i915_error_state_buf *m,
 	}
 	err_printf(m, "  seqno: 0x%08x\n", ring->seqno);
 	err_printf(m, "  waiting: %s\n", yesno(ring->waiting));
-
-	err_printf(m, "  ring->ctx: %p\n", ring->ring_context);
-	err_printf(m, "  ring->size: 0x%08x\n", (unsigned int) ring->size);
-	err_printf(m, "  ring->space: 0x%08x\n", (unsigned int) ring->space);
-	err_printf(m, "  ring->last_retired_head: 0x%08x\n", ring->last_retired_head);
 	err_printf(m, "  ring->head: 0x%08x\n", ring->cpu_ring_head);
 	err_printf(m, "  ring->tail: 0x%08x\n", ring->cpu_ring_tail);
 	err_printf(m, "  hangcheck: %s\n",
@@ -421,12 +416,10 @@ int i915_error_state_to_str(struct drm_i915_error_state_buf *m,
 				   dev_priv->ring[i].name,
 				   error->ring[i].num_requests);
 			for (j = 0; j < error->ring[i].num_requests; j++) {
-				err_printf(m, "  seqno 0x%08x, emitted %ld, tail 0x%08x [head 0x%x ctx %p]\n",
+				err_printf(m, "  seqno 0x%08x, emitted %ld, tail 0x%08x\n",
 					   error->ring[i].requests[j].seqno,
 					   error->ring[i].requests[j].jiffies,
-					   error->ring[i].requests[j].tail,
-					   error->ring[i].requests[j].head,
-					   error->ring[i].requests[j].ctx);
+					   error->ring[i].requests[j].tail);
 			}
 		}
 
@@ -984,10 +977,6 @@ static void i915_gem_record_rings(struct drm_device *dev,
 		} else
 			rbuf = ring->buffer;
 
-		error->ring[i].ring_context = rbuf->ctx;
-		error->ring[i].size = rbuf->size;
-		error->ring[i].space = rbuf->space;
-		error->ring[i].last_retired_head = rbuf->last_retired_head;
 		error->ring[i].cpu_ring_head = rbuf->head;
 		error->ring[i].cpu_ring_tail = rbuf->tail;
 
@@ -1017,11 +1006,9 @@ static void i915_gem_record_rings(struct drm_device *dev,
 			struct drm_i915_error_request *erq;
 
 			erq = &error->ring[i].requests[count++];
-			erq->ctx = request->ctx;
-			erq->head = request->head;
-			erq->tail = request->tail;
 			erq->seqno = request->seqno;
 			erq->jiffies = request->emitted_jiffies;
+			erq->tail = request->tail;
 		}
 	}
 }
-- 
1.7.9.5

